### ISSUE TITLE
addFilter in the initialization of Viz

### ISSUE THREAD

**Issue by jontom33** [tableau/embedding-api-v3-guide#21]
I'm trying to add a filter in the initialization of a Viz using the viz.addFilter but it is not working.

import {TableauViz} from '../tableau.embedding.3.0.0-alpha.23.js'

let viz = new TableauViz();

viz.src = 'http://my-server/views/my-workbook/my-view';
viz.toolbar = 'hidden';
viz.addFilter('Region', 'Central');
viz.addEventListener('marksSelected', handleMarksSelection);

Is this not available in 3.0 or when will it be available.

Thanks.

**Comment by illonage** [comment]
Hi @jontom33 - Initialization via JavaScript is going to be available in v3.1, you can already try it out in Developer Preview: https://embedding.tableauusercontent.com/preview/getting-started-v3.html

Let us know if it is working for you

**Comment by jontom33** [comment]
I couldn't get filtering or a firstInteractive listener working.  Is that going to be available in 3.1?

**Comment by jmore91hca** [comment]
I was able to get filtering working on initialization by adding the viz-filter element before appending to the div.


``` JavaScript
const viz = new TableauViz();

viz.src = "http://my-server/views/my-workbook/my-view";
viz.toolbar = 'hidden';
viz.hideTabs = 'true';
let filter = document.createElement("viz-filter");
filter.setAttribute('field', 'Name');
filter.setAttribute('value', 'Test');

viz.appendChild(filter);
containerDiv.appendChild(viz);
```

**Comment by illonage** [comment]
@jontom33 in v3, you can follow these instructions [here](https://help.tableau.com/current/api/embedding_api/en-us/docs/embedding_api_filter.html) for filtering. Let us know if you have comment on the documentation.

**Comment by ashwinar** [comment]
@jontom33 Viz.addFilter() will be available in Embedding API v3.2. It's currently in preview, if you'd like to go check it out.
